src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleRequestSender.cs

```diff
@@ -770,5 +778,33 @@ private void HandleCustomHostLaunch(ITestHostLauncher customHostLauncher, Messag
                 this.communicationManager.SendMessage(MessageType.CustomTestHostLaunchCallback, ackPayload, this.protocolVersion);
             }
         }
+
+        private void AttachDebuggerToProcess(ITestHostLauncher customHostLauncher, Message message)
+        {
+            if (customHostLauncher == null)
+            {
+                throw new ArgumentNullException("VsTestConsoleRequestSender.AttachDebuggerToProcess: test host launcher cannot be null.");
+            }
+            if (!(customHostLauncher is IDebugTestHostLauncher launcher))
+            {
+                throw new NotSupportedException("VsTestConsoleRequestSender.AttachDebuggerToProcess: test host launcher is not ITestHostLauncher2.");
+            }
+
+            bool response = false;
+            try
+            {
+                var pid = this.dataSerializer.DeserializePayload<int>(message);
+                response = launcher.AttachDebuggerToProcess(pid);
```

Comment:
Looks like the response in case of `HandleCustomHostLaunch` is actually a type that can also specify an `ErrorMessage` (in case exception is caught). Perhaps we should do the same here as well?

Reply:
There is a comment about special abort handling in vstest.console.exe when `ErrorMessage` is specified. Please take a look as we may need the same handling here to preserve what happens when the attach fails.

Reply:
Done.